[macOS] Safari occasionally hangs when right clicking an image

pxlcoder opened the pull request:
#### 1b8516049c4998a8f926a2a73e2b60fbdaea8002
<pre>
[macOS] Safari occasionally hangs when right clicking an image
<a href="https://bugs.webkit.org/show_bug.cgi?id=256720">https://bugs.webkit.org/show_bug.cgi?id=256720</a>
rdar://108912950

Reviewed by Wenson Hsieh.

WebKit performs QR code detection on behalf of Safari when right clicking an
image. `-[VNImageRequestHandler performRequests:error:]` is used to perform
detection. While each request has a completion handler, the call to perform
requests is actually synchronous. Since context menu presentation is already
asynchronous, WebKit can and should perform detection on a background thread to
avoid hangs.

Covered by existing API tests in ContextMenuTests.

* Source/WebKit/Platform/cocoa/ImageAnalysisUtilities.mm:
(WebKit::requestPayloadForQRCode):

Perform QR code detection on a `WorkQueue`.

Canonical link: <a href="https://commits.webkit.org/264025@main">https://commits.webkit.org/264025@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/879dab4c24a7e20ad01c6c2290bbc85c13459e51

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6445 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6653 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/6834 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8024 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6740 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/6955 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6604 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/9634 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6558 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/6601 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/5836 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8103 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/4066 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/13671 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/5987 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/5896 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/8202 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6381 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5217 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5787 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1523 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9946 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6158 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
lgtm. ~~Can you fix the style error?~~

pxlcoder committed:
[macOS] Safari occasionally hangs when right clicking an image
https://bugs.webkit.org/show_bug.cgi?id=256720
rdar://108912950

Reviewed by Wenson Hsieh.

WebKit performs QR code detection on behalf of Safari when right clicking an
image. `-[VNImageRequestHandler performRequests:error:]` is used to perform
detection. While each request has a completion handler, the call to perform
requests is actually synchronous. Since context menu presentation is already
asynchronous, WebKit can and should perform detection on a background thread to
avoid hangs.

Covered by existing API tests in ContextMenuTests.

* Source/WebKit/Platform/cocoa/ImageAnalysisUtilities.mm:
(WebKit::requestPayloadForQRCode):

Perform QR code detection on a `WorkQueue`.

Canonical link: https://commits.webkit.org/264025@main
